app/Http/Controllers/Service/CentreController.php

```diff
@@ -34,4 +35,154 @@ public function printRegistrations(Centre $centre)
+                switch ($status['eligibility']) {
+                    case 'Pregnancy':
+                        $due_date = $child->dob->format('d/m/Y');
+                        break;
+                    case 'Eligible':
+                        $kids[$dob_header] = $child->dob->format('m/Y');
+                        $eligible += 1;
+                        break;
+                    case "Ineligible":
+                        $kids[$dob_header] = $child->dob->format('m/Y');
+                        break;
+                }
+            }
+            // Add count oif eligible kids
+            $row["Eligible Children"] = $eligible;
+
+            // Add our kids back in
+            $row=array_merge($row, $kids);
+
+            // TODO: What happens for families with two concurrent pregnancies?
+            $row["Due Date"] = $due_date;
+            $row["Leaving Date"] = null;
+
+            // update the headers if necessary
+            if (count($headers) < count($row)) {
+                $headers = array_keys($row);
+            }
+            // stack new row onto the array
+            $rows[] = $row;
+        }
+
+        // en-sparsen the rows with empty fields for unused header.
+        foreach ($rows as $index => $row) {
+            $sparse_row = [];
+            foreach ($headers as $header) {
+                $sparse_row[$header] = (array_key_exists($header, $row)) ? $row[$header] : null;
+            }
+            // Key/value order matters to laravel-excel
+            $rows[$index] = $sparse_row;
+        }
+
+        /**
+         * TODO: write an OO system for formatting things better.
+         * Ideally we'd have formatting for
+         * - rows with a leaving date showing grey
+         * - ineligible children showing grey
+         * - children with changes in near future showing red.
+         */
+
+        $excel_doc = Excel::create(
+            'RegSummary_' . $now->format('YmdHis'),
+            function ($excel) use ($user, $rows, $headers) {
+                $excel->setTitle('Registration Summary');
+                $excel->setDescription('Summary of Registrations from Centres available to '. $user->name);
+                $excel->setManager($user->name);
+                $excel->setCompany(env('APP_URL'));
+                $excel->setCreator(env('APP_NAME'));
+                $excel->setKeywords([]);
+                $excel->sheet(
+                    'Registrations',
+                    function ($sheet) use ($rows, $headers) {
+                        $sheet->setOrientation('landscape');
+                        $sheet->row(1, $headers);
+                        $sheet->cells('A1', function ($cells) {
+                            $cells->setBackground('#6495ED')
+                                ->setFontWeight('bold');
+                        });
+                        $letters = range('A', 'Z');
+                        $sheet->cells('B1:' . $letters[count($headers)-1] .'1', function ($cells) {
```

Comment:
Again would be easy if you just started with a fixed number.

Reply:
Would comma or pipe separated kids be ok?

Reply:
I'm trying to match a specific sheet as closely as possible to keep the FM guys life consistent as possible.